src/components/BivariateLegend/BivariateLegend.tsx

```diff
@@ -115,3 +114,29 @@ export function BivariateLegend({
     </div>
   );
 }
+
+const formatBivariateAxisLabel = (quotients: Axis['quotients']): string => {
+  if (!quotients) return '';
+  const [nominator, denominator] = quotients;
+
+  // we don't have nominator unit - don't show units
+  if (isNilOrOther(nominator.unit.id)) {
+    return `${nominator.label} to ${denominator.label}`;
+  }
+  // we have nominator unit and denominator unit is - one
+  if (denominator.name === 'one') {
+    return `${nominator.label} (${nominator.unit.shortName})`;
+  }
+  // we have nominator unit, but don't have denominator unit - show only nominator unit
+  if (isNilOrOther(denominator.unit.id)) {
+    return `${nominator.label} to ${denominator.label} (${nominator.unit.shortName})`;
+  }
+  // here we process cases where we have both units
+  if (denominator.name === 'area_km2') {
+    return `${nominator.label} (${nominator.unit.shortName}/${denominator.unit.shortName})`;
+  }
+
+  return `${nominator.label} to ${denominator.label} (${nominator.unit.shortName}/${denominator.unit.shortName})`;
+};
+
+const isNilOrOther = (unitId: string | null): boolean => !unitId || unitId === 'other';
```

Comment:
since we're implementing new test coverage policy that function looks like a candidate for a simple test

Reply:
ok, will add!

Reply:
done